File: src/main/java/p1003.java
import java.util.ArrayList;
import java.util.Scanner;

public class p1003 {
	
	public static void main(String[] args) {
		Scanner scanner = new Scanner(System.in);

		int testCase = scanner.nextInt();
		int[] result = new int[testCase];
		ArrayList<Integer> zeroResult = new ArrayList<Integer>();
		ArrayList<Integer> oneResult = new ArrayList<Integer>();

		zeroResult.add(1);
		zeroResult.add(0);
		oneResult.add(0);
		oneResult.add(1);

		int setNum = 2;

		while(setNum != 41) {
			zeroResult.add(zeroResult.get(setNum - 1) + zeroResult.get(setNum - 2));
			oneResult.add(oneResult.get(setNum - 1) + oneResult.get(setNum - 2));
			setNum++;
		}

		for(int i = 0; i < testCase; i++) {
			result[i] = scanner.nextInt();
		}

		for(int i = 0; i < testCase; i++) {
			System.out.println(zeroResult.get(result[i]) + " " + oneResult.get(result[i]));
		}

		scanner.close();
	}
}

File: C++/C++/p1003.cpp
//
//  p1003.cpp
//  C++
//
//  Created by KimHanJu on 09/09/2019.
//  Copyright © 2019 KimHanJu. All rights reserved.
//

#include <iostream>
#include <vector>
using namespace std;

int main(int argc, const char * argv[]) {
    vector<int> zeroResult, oneResult;
    
    int testCase;
    cin >> testCase;
    int result[testCase];
    
    zeroResult.push_back(1);
    zeroResult.push_back(0);
    oneResult.push_back(0);
    oneResult.push_back(1);
    
    int setNum = 2;
    
    while(setNum != 41) {
        zeroResult.push_back(zeroResult.at(setNum-1) + zeroResult.at(setNum-2));
        oneResult.push_back(oneResult.at(setNum-1) + oneResult.at(setNum-2));
        setNum++;
    }
    
    for(int i = 0; i < testCase; i++) cin >> result[i];
    
    for(int i = 0; i < testCase; i++) cout << zeroResult.at(result[i]) << " " << oneResult.at(result[i]) << "\n";
    
    return 0;
}
